### PR TITLE
Bump dependency versions (resin-docker-build and a few others)

### PR DESCRIPTION
**Pull request by pdcastro**
Change-type: patch
Signed-off-by: Paulo Castro <paulo@balena.io>

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -39,19 +39,19 @@
...
+    "bluebird": "^3.5.3",
+    "docker-progress": "^3.0.4",
     "docker-toolbelt": "^3.3.7",
     "dockerode": "^2.5.5",
     "lodash": "^4.17.4",
     "resin-bundle-resolve": "^3.0.1",
-    "resin-compose-parse": "^2.0.1",
-    "resin-docker-build": "^1.0.0",
+    "resin-compose-parse": "^2.0.3",
+    "resin-docker-build": "^1.0.1",
```
Comment (pdcastro) [inline package.json]: `resin-docker-build` is the important change here, to incorporate this PR: https://github.com/balena-io/resin-docker-build/pull/31
The other packages were upgraded to match `resin-builder`.

File: package.json [hunk 2]
```diff
@@ -39,19 +39,19 @@
     "typescript": "^2.6.2"
   },
   "dependencies": {
-    "@types/bluebird": "3.5.18",
+    "@types/bluebird": "3.5.20",
     "@types/dockerode": "2.5.5",
-    "@types/lodash": "^4.14.87",
-    "@types/tar-stream": "^1.6.0",
+    "@types/lodash": "4.14.112",
+    "@types/tar-stream": "1.6.0",
```
Comment (pdcastro) [inline package.json]: Pinning `@types` packages to specific versions (removing the caret prefix): I don't know if this a settled best practice, but I've seen past pull requests aiming at doing specifically that, avoid typescript compilation breaking unexpectedly.

File: package.json [hunk 3]
```diff
@@ -39,19 +39,19 @@
     "typescript": "^2.6.2"
   },
   "dependencies": {
-    "@types/bluebird": "3.5.18",
+    "@types/bluebird": "3.5.20",
     "@types/dockerode": "2.5.5",
-    "@types/lodash": "^4.14.87",
-    "@types/tar-stream": "^1.6.0",
+    "@types/lodash": "4.14.112",
```
Comment (zvin) [inline package.json]: Why did you remove the `^`?

Comment (pdcastro) [reply package.json]: Because of this pull request from @Page- : https://github.com/balena-io/resin-builder/pull/502
Is this best practice or not?...
